plugins/statistics.koplugin/main.lua

```diff
@@ -234,28 +266,138 @@ function getDatesForBookOldFormat(book)
     return generateReadBooksTable(book.title, dates)
 end
 
-function getDatesForBook(book)
+-- Use from here:
+-- http://stackoverflow.com/questions/1340230/check-if-directory-exists-in-lua/21637668#21637668
+function isDir(name)
+    if type(name)~="string" then return false end
+    local cd = lfs.currentdir()
+    local is = lfs.chdir(name) and true or false
+    lfs.chdir(cd)
+    return is
+end
+
+
+-- sdays -> number of days to show
+-- ptype -> daily - show daily without weekday name
+--          daily_weekday - show daily with weekday name
+--          weekly - show weekly
+--          monthly - show monthly
+function getDatesFromAll(sdays, ptype)
     local dates = {}
+    local sorted_tbl = {}
+    local diff
+    local book = {}
+    -- read from history file
+    local result_history = ReaderStatistics:readHistoryFile()
+    -- now timestamp
+    now_t = os.date("*t")
+    local from_begin_day = now_t.hour *3600 + now_t.min*60 + now_t.sec
+    local now_stamp = os.time()
+    local one_day = 24 * 3600 -- one day in seconds
+        for k, v in pairs(result_history) do
+            path_h,file_h,extension_h = string.match(v.file, "(.-)([^/]-([^/%.]+))$")
+            local file_without_ext = string.match(file_h, "(.+)%..+")
+            local path_sdr = path_h .. file_without_ext .. ".sdr"
```

Comment:
Actually, if all you want is to read the setting for the book, you can just use `DocSettings:open(v.f)`. This method also handles support for legacy setting formats.